index.js

```diff
@@ -7,33 +7,109 @@ function ask(questionText) {
   });
 }
 
-// remember the StateMachine lecture
-// https://bootcamp.burlingtoncodeacademy.com/lessons/cs/state-machines
-let states = {
-  'roomOne': { canChangeTo: [ 'roomTwo' ] },
-  'roomTwo': { canChangeTo: [ 'roomThree' ] },
-  'roomThree': { canChangeTo: [ 'roomOne' ] }
-};
-
-let currentState = "green";
-
-function enterState(newState) {
-  let validTransitions = states[currentState].canChangeTo;
-  if (validTransitions.includes(newState)) {
-    currentState = newState;
-  } else {
-    throw 'Invalid state transition attempted - from ' + currentState + ' to ' + newState;
-  }
-}
 
 start();
+let room1 = {
+  name: "182 Main St",
+  inventory: ['Sign'],
+  message: "Your are at 182 Main St"
+
+}
+let room2 = {
+  name: "Foyer",
+  inventory: ['News Paper'],
+  message: "You have entered the Foyer"
+}
+let room3 = {
+  name: "Classroom",
+  inventory: ['Pencil'],
+  message: "You have entered the Classroom, are you ready for the lesson?"
+}
+let room4 = {
+  name: "Muddy Waters",
+  inventory: ['Coffee'],
+  message: "Welcome to Muddy Waters"
 
+}
+let room5 = {
+  name: "Mr. Mikes",
+  inventory: ['Pizza'],
+  message: "Welcome to Mr. Mikes"
+}
+let states = {
+  "182 Main St": { canChangeTo: ["Foyer", "Muddy Waters", "Mr. Mikes"] },
+  "Foyer": { canChangeTo: ["Classroom", "182 Main St"] },
+  "Classroom": { canChangeTo: ["Foyer"] },
+  "Mudddy Waters": { canChangeTo: ["182 Main St"] },
+  "Mr. Mikes": { canChangeTo: ["182 Main St"] }
+}
+let currentRoom = room1;
+let userInventory = []
 async function start() {
   const welcomeMessage = `182 Main St.
 You are standing on Main Street between Church and South Winooski.
-There is a door here. A keypad sits on the handle.
-On the door is a handwritten sign.`;
+There is a door here. A keypad sits on the handle.\nOn the door is a hand written message.`
+
   let answer = await ask(welcomeMessage);
-  console.log('Now write your code to make this work!');
-  process.exit();
+  while (answer !== 'exit') {
+    answer = await ask('>_')
+    if (answer === 'read message') {
+      console.log("Welcome to Burlington Code Academy! Come up to the third floor. \nIf the door is locked, use the code 12345")
+
+    } else {
+      console.log('Sorry I dont understand ' + answer)
+    }
+
+    let userKeyPadInput = await ask("Now you know the code, enter it now!")
+    while (userKeyPadInput !== '12345') {
+      userKeyPadInput = await ask(">_")
+      if (userKeyPadInput === '12345') {
+        moveToRoom(room2)
+        console.log(room2.message)
+        console.log("There is a Newspaper in the corner")
+      }
+    }
+
+    let userMove = await ask("The classroom is up stairs. You can go to class.\nOr go to main.")
```

Comment:
I would recommend you only have one variable for user input. Setting up multiple variables makes your code harder to follow as you have to keep track of not only where you are, but also which variable controls which loop, and what the current values of all the variables are.